numpy/core/arrayprint.py

```diff
@@ -1076,6 +1076,37 @@ def format_float_positional(x, precision=None, unique=True,
                               sign=sign, pad_left=pad_left,
                               pad_right=pad_right)
 
+# currently this is hidden from users, access using
+# >>> from np.core.arrayprint import format_float_OSprintf_g
+def format_float_OSprintf_g(val, precision):
+    """
+    Print a floating point scalar using the system's printf function,
+    equivalent to:
+
+        printf("%.*g", precision, val);
+
+    for half/float/double, or replacing 'g' by 'Lg' for longdouble. This
+    method is designed to help cross-validate the format_float_* methods.
+
+    Parameters
+    ----------
+    val : python float or numpy floating scalar
+        Value to format.
+
+    precision : non-negative integer, optional
+        Precision given to printf.
+
+    Returns
+    -------
+    rep : string
+        The string representation of the floating point value
+
+    See Also
+    --------
+    format_float_scientific
+    format_float_positional
+    """
+    return printf_float_g(val, precision)
```

Comment:
Why the python wrapper?

Reply:
Just for the docstring

Reply:
Precedent seems to be to use `add_new_docs` for that.